src/components/drag-and-drop-list/DragAndDropList.tsx

```diff
@@ -0,0 +1,137 @@
+import {useRef, useState} from "react";
+import "./drag-and-drop-list.css";
+import {DragAndDropTodoItem} from "../drag-and-drop-todo-item/DragAndDropTodoItem";
+import {GhostItem} from "../ghost-item/GhostItem";
+
+interface DragAndDropListProps {
+  title: string
+  listItems: Array<string>
+  listId: number
+  setListItems: (listItems: Array<string>) => void;
+  onItemDropped?: (item: string, sourceListId: number, targetListId: number) => void;
+}
+interface Coordinates {
+  x: number
+  y: number
+}
+
+export function DragAndDropList(props: DragAndDropListProps) {
+  const { title, listItems: items, setListItems: setItems, listId, onItemDropped } = props;
+
+  const [dragStartIdx, setDragStartIdx] = useState<number | null>(null);
+  const [isDragging, setIsDragging] = useState(false);
+  const [ghostPosition, setGhostPosition] = useState<Coordinates>({ x: 0, y: 0 });
+  const [draggedItem, setDraggedItem] = useState<string | null>(null);
+  const [hoverIndex, setHoverIndex] = useState(null);
+  const [isOverDraggableArea, setIsOverDraggableArea] = useState(false);
```

Comment:
Looks like its better to start with zero, so that you won't have to check for null values afterwards. (splice for example, it does not accept null values. If you start with null, then you would have to do if(hoverIndex) etc.)
```suggestion
  const [hoverIndex, setHoverIndex] = useState(0);
```